index.js

```diff
@@ -237,7 +237,14 @@ module.exports = {
       };
 
       self.exportRecord = function (req, piece, record, callback) {
-        const schema = self.schema;
+        let schema = self.schema;
+        if (self.options.exportOmitProperties) {
+          for (let prop of self.options.exportOmitProperties) {
```

Comment:
This seems like double looping. Maybe using `delete` so you don't need the extra `.filter`? Otherwise it looks totally workable.

Reply:
I'd still have to loop through schema to use delete - maybe I'm misunderstanding?

Reply:
yes, this is correct but very inefficient, you're making a lot of copies of the array and you're doing it on every pass for every new record. I would still use `filter`, but in a single pass, using `includes` to check whether each schema field's name is present in `omitFields` or not. And I'd do this earlier, in a method called from `afterConstruct`, and attach the result to `self` as `self.exportSchema`.